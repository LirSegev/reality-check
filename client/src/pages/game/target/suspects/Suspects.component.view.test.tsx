import { render, cleanup, fireEvent } from '@testing-library/react';
import React from 'react';

import Suspects from './Suspects.component.view';
import styles from './Suspects.module.css';

afterEach(cleanup);

it('renders', () => {
	const { asFragment } = render(<Suspects showId={1} handleClick={() => {}} />);

	expect(asFragment()).toMatchSnapshot();
});

it('shows correct suspect', () => {
	const { asFragment } = render(<Suspects showId={2} handleClick={() => {}} />);

	const shownImages = asFragment().querySelector(`.${styles.show}`);
	expect((shownImages as HTMLElement).dataset.suspect_id).toBe('2');
});

it('shows only one image at a time', () => {
	const { asFragment } = render(<Suspects showId={2} handleClick={() => {}} />);

	const shownImages = asFragment().querySelectorAll(`.${styles.show}`);
	expect(shownImages).toHaveLength(1);
});

it('sets data-suspect-id for images', () => {
	const { asFragment } = render(<Suspects showId={1} handleClick={() => {}} />);

	const shownImages = asFragment().querySelector(`.${styles.show}`);
	expect((shownImages as HTMLElement).dataset).toHaveProperty('suspect_id');
});

it('calls handleClick() correctly', () => {
	const handleClick = jest.fn().mockImplementation(e => {
		expect(e.currentTarget.dataset.suspect_id).toBe('1');
	});
	const { container } = render(
		<Suspects showId={1} handleClick={handleClick} />
	);

	fireEvent.click(container.querySelector(`.${styles.show}`)!);

	expect(handleClick).toBeCalled();
});

it('shows default image when showId is undefined', () => {
	const { getByTestId } = render(
		<Suspects showId={undefined} handleClick={() => {}} />
	);

	expect(getByTestId('default')).toHaveClass(styles.show);
});
