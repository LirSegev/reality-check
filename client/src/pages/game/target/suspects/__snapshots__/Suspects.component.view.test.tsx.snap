// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    id="suspects"
  >
    <div
      id="images"
    >
      <img
        alt=""
        data-testid="default"
        src="/images/suspects/default.jpg"
      />
      <img
        alt=""
        data-suspect_id="0"
        src="/images/suspects/0.jpg"
      />
      <img
        alt=""
        class="show"
        data-suspect_id="1"
        src="/images/suspects/1.jpg"
      />
      <img
        alt=""
        data-suspect_id="2"
        src="/images/suspects/2.jpg"
      />
      <img
        alt=""
        data-suspect_id="3"
        src="/images/suspects/3.jpg"
      />
      <img
        alt=""
        data-suspect_id="4"
        src="/images/suspects/4.jpg"
      />
      <img
        alt=""
        data-suspect_id="5"
        src="/images/suspects/5.jpg"
      />
      <img
        alt=""
        data-suspect_id="6"
        src="/images/suspects/6.jpg"
      />
      <img
        alt=""
        data-suspect_id="7"
        src="/images/suspects/7.jpg"
      />
      <img
        alt=""
        data-suspect_id="8"
        src="/images/suspects/8.jpg"
      />
      <img
        alt=""
        data-suspect_id="9"
        src="/images/suspects/9.jpg"
      />
      <img
        alt=""
        data-suspect_id="10"
        src="/images/suspects/10.jpg"
      />
      <img
        alt=""
        data-suspect_id="11"
        src="/images/suspects/11.jpg"
      />
      <img
        alt=""
        data-suspect_id="12"
        src="/images/suspects/12.jpg"
      />
      <img
        alt=""
        data-suspect_id="13"
        src="/images/suspects/13.jpg"
      />
      <img
        alt=""
        data-suspect_id="14"
        src="/images/suspects/14.jpg"
      />
      <img
        alt=""
        data-suspect_id="15"
        src="/images/suspects/15.jpg"
      />
      <img
        alt=""
        data-suspect_id="16"
        src="/images/suspects/16.jpg"
      />
      <img
        alt=""
        data-suspect_id="17"
        src="/images/suspects/17.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
